src/validateChangelog.ts

```diff
@@ -0,0 +1,117 @@
+import { Version } from './constants';
+import { parseChangelog } from './parseChangelog';
+
+/**
+ * Indicates that the changelog is invalid.
+ */
+export class InvalidChangelogError extends Error {}
+
+/**
+ * Indicates that unreleased changes are still present in the changelog.
+ */
+export class UnreleasedChangesError extends InvalidChangelogError {
+  constructor() {
+    super('Unreleased changes present in the changelog');
+  }
+}
+
+/**
+ * Indicates that the release header for the current version is missing.
+ */
+export class MissingCurrentVersionError extends InvalidChangelogError {
+  /**
+   * @param currentVersion - The current version
+   */
+  constructor(currentVersion: Version) {
+    super(`Current version missing from changelog: '${currentVersion}'`);
+  }
+}
+
+/**
+ * Represents a formatting error in a changelog.
+ */
+export class ChangelogFormattingError extends InvalidChangelogError {
+  public data: Record<string, string>;
```

Comment:
Curious that in `cli.ts`, when this is referenced, it's able to infer that it has entries for `validChangelog` and `invalidChangelog` :thinking: I don't quite follow how it knows those properties exist. Some kinda fancy type inference maybe.